### DISCUSSION TITLE
How much GPU memory does Whipser large-v3 fine-tuning require?

### DISCUSSION THREAD

**Discussion by toanhuynhnguyen** [openai/whisper#2397] [Q&A]
Hi everyone,
I follow this document: https://huggingface.co/blog/fine-tune-whisper
How much GPU memory does Whipser large-v3 fine-tuning require?
Thank you.

**Comment by phineas-pta** [comment]
at least 24gb

**Reply by toanhuynhnguyen** [reply]
Thanks bro, I already tested on Colab, it takes around 32GB.

**Reply by toanhuynhnguyen** [reply]
I fine-tuned small and large-v3 models as that document:
- Fine-tuned small model works normally.
- Fine-tubed large-v3 does not work normally, when I transcribe a Chinese audio then it auto-translates to English though I specified task="transcribe". Can you please give me some advice on this issue? Thanks.

**Reply by phineas-pta** [reply]
bác có ép lang+task lúc fine tune ko, lúc chạy inference cũng thế

**Reply by toanhuynhnguyen** [reply]
Mình làm theo notebook này của bài viết trên https://colab.research.google.com/github/sanchit-gandhi/notebooks/blob/main/fine_tune_whisper.ipynb#scrollTo=77d9f0c5-8607-4642-a8ac-c3ab2e223ea6
Mình thay ngôn ngữ Hindi trong notebook bằng ngôn ngữ mình cần fine-tune ví dụ: Chinese, Malay. Còn lại mình không thay đổi gì hết của notebook. Vậy là "có ép lang+task" như bạn nói không?

Khi chạy inference: Ví dụ audio Chinese thì mình set lang là "zh", audio tiếng Malay thì mình set lang là "ms". Còn mình set task="transcribe". Không biết nó bị lỗi gì mà khi chạy inference nó tự động translate sang tiếng Anh hết. Mong bạn cho mình lời khuyên, thanks bạn nhiều.

**Reply by phineas-pta** [reply]
bác fine tune cả 2 cùng lúc hay sao, nếu fine tune lần lượt thì whisper gần như bị phế, lúc fine tune xong bác có check lại WER hay ko

**Reply by toanhuynhnguyen** [reply]
Mình chỉ fine-tune 1 ngôn ngữ rồi mình dùng model sau khi đã fine-tune chạy luôn chứ ko fine-tune 2 ngôn ngữ cùng lúc.
Ví dụ: từ original large-v3 model mình fine-tune cho tiếng Malay, sau đó dùng model fine-tune này để chạy inference cho audio tiếng Malay luôn nhưng nó tự động translate sang tiếng Anh, mặc dù mình đã set task="transcribe" và lang="ms".

Đây là WER của nó:
![image](https://github.com/user-attachments/assets/d7763d71-e49b-4dfb-a530-73eb8051bae4)

**Reply by phineas-pta** [reply]
WER này bác có kiểm lại ko, có bị dịch tiếng anh ko

**Reply by toanhuynhnguyen** [reply]
À để mình check lại nhé, cảm ơn bạn rất nhiều.

**Reply by toanhuynhnguyen** [reply]
Mình vừa test lại như bạn nói, kết quả khá kỳ lạ.
Nếu mình set task="transcribe" thì nó auto-translate sang tiếng Anh, còn khi set task="translate" thì nó lại transcribe như bình thường ko auto-translate. Không hiểu nổi :(
